controller/linodemachine_controller.go

```diff
@@ -675,16 +693,10 @@ func (r *LinodeMachineReconciler) reconcileDelete(
 		return nil
 	}
 
-	if machineScope.LinodeCluster.Spec.Network.LoadBalancerType != "dns" {
-		if err := services.DeleteNodeFromNB(ctx, logger, machineScope); err != nil {
-			logger.Error(err, "Failed to remove node from Node Balancer backend")
-			return err
-		}
-	} else {
-		if err := services.DeleteIPFromDNS(ctx, machineScope); err != nil {
-			logger.Error(err, "Failed to remove IP from DNS")
-			return err
-		}
+	if err := r.removeMachineFromLB(ctx, logger, machineScope); err != nil {
+		logger.Error(err, "Failed to remove machine from LB")
+
+		return err
```

Comment:
Please only log or return the error. There is seldom reason to do both.

If you choose to return the error here, please wrap it like so:

```go
return fmt.Errorf("remove machine from loadbalancer: %w", err)
```

Reply:
done